### PR TITLE
Unread msg management

### PR DESCRIPTION
**Pull request by ylecollen**

### CODE REVIEW

**Comment by ylecollen** [comment]
one case the read marker should not be displayed
-> scroll back in the room history
-> receive one new event
-> click on the red back arrow to display the latest messages

A read marker is displayed after the latest event

**Comment by ylecollen** [comment]
Another case
Open the room activity
put the app in background
send > 10 messages in the room

The read marker is not displayed at the right position

**Comment by ylecollen** [comment]
Another case
Scroll to back in the history
Receive 5 new messages 
scroll down

The read marker is only displayed when the history reaches the latest event

File: vector/src/main/java/im/vector/activity/VectorRoomActivity.java [hunk 1]
```diff
@@ -927,9 +968,48 @@ public void onDestroy() {
...
+                mRoom.setReadMakerEventId(newReadMarkerEventId);
+            }
+        }
+    }
+
     @Override
     protected void onPause() {
         super.onPause();
+
+        if (mVectorMessageListFragment.getMessageAdapter().isUnreadViewMode()) {
```
Comment (sabrinaj) [inline vector/src/main/java/im/vector/activity/VectorRoomActivity.java]: Done

File: vector/src/main/java/im/vector/activity/VectorRoomActivity.java [hunk 2]
```diff
@@ -1202,6 +1283,118 @@ public void onScroll(int firstVisibleItem, int visibleItemCount, int totalItemCo
...
+        boolean hasUnreadMessageHidden = false;
+        RoomSummary summary = mRoom.getDataHandler().getStore().getSummary(mRoom.getRoomId());
+        if (summary != null) {
+            String readMarkerEventId = summary.getReadMarkerEventId();
+            String readReceiptEventId = summary.getReadReceiptEventId();
+            Log.d(LOG_TAG, "## checkUnreadMessage readMarkerEventId " + readMarkerEventId+" readReceiptEventId "+readReceiptEventId);
+
+            synchronized (this){
+                if (readMarkerEventId != null && !readMarkerEventId.equals(readReceiptEventId)) {
+                    if (readMarkerEventId.startsWith(mRoom.getRoomId())) {
```
Comment (sabrinaj) [inline vector/src/main/java/im/vector/activity/VectorRoomActivity.java]: Done

**Pull request closed without merging**